docling_sdg/qa/conceptual_generate.py

```diff
@@ -0,0 +1,427 @@
+        start_time = time.time()
+
+        topics = self.generate_topics(content_description)
+
+        result = self.generate_questions_from_topics(content_description, topics)
+        end_time = time.time()
+        result.time_taken = end_time - start_time
+
+        return result
+
+    def chat(self, message: str) -> str:
+        messages = [ChatMessage(role="user", content=message)]
+        resp = self.llm.chat(messages)
+        block = resp.message.blocks[0]
+        if isinstance(block, TextBlock):
+            response_text = block.text
+        else:
+            _log.warning(f"Unexpected block type: {type(block)}")
+            response_text = ""
+        return response_text
+
+    def generate_topics(self, content_description: str) -> dict[UserProfile, list[str]]:
+        user_profiles = self.options.user_profiles
+        topic_prompts = self.options.topic_prompts
+        profiles_with_topics: dict[UserProfile, list[str]] = {}
+        for user_profile in user_profiles:
+            for prompt in topic_prompts:
+                message = prompt.format(
+                    content_description_str=content_description,
+                    num_topics=user_profile.number_of_topics,
+                    user_profile_str=user_profile.description,
+                )
+                response_text = self.chat(message)
+                topic_list = _extract_list_items(response_text)
+                if user_profile not in profiles_with_topics:
+                    profiles_with_topics[user_profile] = topic_list
+                else:
+                    profiles_with_topics[user_profile].extend(topic_list)
+        return profiles_with_topics
+
+    def generate_questions_from_topics(
+        self,
+        content_description: str,
+        profiles_with_topics: dict[UserProfile, list[str]],
+    ) -> GenerateResult:
+        question_prompts = self.options.question_prompts
+        additional_instructions = self.options.additional_instructions
+        start_time = time.time()
+        num_questions_expected = _compute_num_questions_expected(
+            profiles_with_topics, question_prompts
+        )
+        i = 0
+
+        # If the file already exists, we remove it so that we don't get duplicate
+        # questions.  Note that this is different from the behavior of the
+        # generate_questions_from_chunks method in generate.py, which allows you to
+        # resume generation from the last question that was generated if it fails.
+        # That is a nice feature, but it would be more difficult to implement for
+        # conceptual generation because we don't have chunk IDs.  In principle, we
+        # could produce IDs for user_profile, topic, and question_prompt, and iteration
+        # number tuples, but it would be more complex to implement, and it is not clear
+        # that it is needed because question generation is fairly lightweight.
```

Comment:
I ran into this issue when offering docling-sdg functionality in a hosted service - sometimes other parameters changed, such as how many chunks to use for QA gen in my service or a hand changed prompt, and I had to add code to remove the cached file.

The simplest solution might be to add something like an `invalidate_cache` flag to `GenerateOptions`, which this would forcibly set to true.

Reply:
I like the invalidate_cache flag idea.